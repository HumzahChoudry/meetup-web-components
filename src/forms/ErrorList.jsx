import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';

export const ERROR_CLASSNAME = 'text--error text--small';

const ErrorList = (props) => {
	const {
		errorId,
		errors,
		...other
	} = props;

	const errorList = Array.isArray(errors) ? errors : [errors];

	if (errorId) {
		other['id'] = errorId;
	}

	return (
		<ul
			aria-live="assertive"
			{...other}
		>
			{
				errorList.map((err, key) => (
					<li key={key} className={cx(ERROR_CLASSNAME, 'text--small')}>{err}</li>
				))
			}
		</ul>
	);
};

ErrorList.propTypes = {
	errorId: PropTypes.string,
	errors: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.array
	])
};

export default ErrorList;
