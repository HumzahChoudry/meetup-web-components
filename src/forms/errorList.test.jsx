import React from 'react';
import ErrorList from './ErrorList';
import { shallow } from 'enzyme';

const MOCK_ERROR = 'This is an error message';
const MOCK_ERROR_ID = 'name';

describe('ErrorList', () => {
	const componentWithId = shallow(
		<ErrorList errorId={MOCK_ERROR_ID} error={MOCK_ERROR} />
	);
	const componentWithoutId = shallow(<ErrorList error={MOCK_ERROR} />);

	it('matches snapshot for single error', () => {
		expect(componentWithId).toMatchSnapshot();
	});
	it('matches snapshot for input without id', () => {
		expect(componentWithoutId).toMatchSnapshot();
	});
});
