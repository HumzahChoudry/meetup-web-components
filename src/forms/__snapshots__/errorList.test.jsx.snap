// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorList matches snapshot for input without id 1`] = `
<ul
  aria-live="assertive"
>
  <li
    className="text--error text--small text--small"
    key="0"
  >
    This is an error message
  </li>
</ul>
`;

exports[`ErrorList matches snapshot for multiple errors 1`] = `
<ul
  aria-live="assertive"
  id="name"
>
  <li
    className="text--error text--small text--small"
    key="0"
  >
    First error message
  </li>
  <li
    className="text--error text--small text--small"
    key="1"
  >
    Second error message
  </li>
  <li
    className="text--error text--small text--small"
    key="2"
  >
    Third error message
  </li>
</ul>
`;

exports[`ErrorList matches snapshot for single error 1`] = `
<ul
  aria-live="assertive"
  id="name"
>
  <li
    className="text--error text--small text--small"
    key="0"
  >
    This is an error message
  </li>
</ul>
`;
